openxr/examples/vulkan.rs

```diff
@@ -0,0 +1,649 @@
+        .collect::<Vec<_>>();
+    println!(
+        "required Vulkan instance extensions: {:?}",
+        vk_instance_exts
+    );
+    let vk_instance_ext_ptrs = vk_instance_exts
+        .iter()
+        .map(|x| x.as_ptr())
+        .collect::<Vec<_>>();
+
+    let vk_entry = ash::Entry::new().unwrap();
+
+    let vk_app_info = vk::ApplicationInfo::builder()
+        .application_version(0)
+        .engine_version(0)
+        .api_version(vk::make_version(1, 1, 0)); // Vulkan 1.1 guarantees multiview support
+
+    let vk_version = vk_entry
+        .try_enumerate_instance_version()
+        .unwrap()
+        .unwrap_or_else(|| vk::make_version(1, 0, 0));
+    let vk_version = xr::Version::new(
+        vk::version_major(vk_version) as u16,
+        vk::version_major(vk_version) as u16,
+        0,
+    );
+
+    let reqs = instance
+        .graphics_requirements::<xr::Vulkan>(system)
+        .unwrap();
+    if reqs.min_api_version_supported > vk_version {
+        panic!(
+            "OpenXR runtime requires Vulkan version > {}",
+            reqs.min_api_version_supported
+        );
+    }
+
+    let vk_device_exts = instance
+        .vulkan_device_extensions(system)
+        .unwrap()
+        .split(' ')
+        .map(|x| CString::new(x).unwrap())
+        .collect::<Vec<_>>();
+    println!("required Vulkan device extensions: {:?}", vk_device_exts);
+    let vk_device_ext_ptrs = vk_device_exts
+        .iter()
+        .map(|x| x.as_ptr())
+        .collect::<Vec<_>>();
+
+    unsafe {
+        let vk_instance = vk_entry
+            .create_instance(
+                &vk::InstanceCreateInfo::builder()
+                    .application_info(&vk_app_info)
+                    .enabled_extension_names(&vk_instance_ext_ptrs),
+                None,
+            )
+            .expect("failed to create Vulkan instance");
+
+        let vk_physical_device = vk::PhysicalDevice::from_raw(
+            instance
+                .vulkan_graphics_device(system, vk_instance.handle().as_raw() as _)
+                .unwrap() as _,
+        );
+
+        let queue_family_index = vk_instance
+            .get_physical_device_queue_family_properties(vk_physical_device)
+            .into_iter()
+            .enumerate()
+            .filter_map(|(queue_family_index, info)| {
+                if info.queue_flags.contains(vk::QueueFlags::GRAPHICS) {
+                    Some(queue_family_index as u32)
+                } else {
+                    None
+                }
+            })
+            .next()
+            .expect("Vulkan device has no graphics queue");
+
+        let vk_device = vk_instance
+            .create_device(
+                vk_physical_device,
+                &vk::DeviceCreateInfo::builder()
+                    .queue_create_infos(&[vk::DeviceQueueCreateInfo::builder()
+                        .queue_family_index(queue_family_index)
+                        .queue_priorities(&[1.0])
```

Comment:
I think this causes issues in release builds

edit: there are similar instances below as well

Reply:
Can you be more specific?

Reply:
Sry! build() will discard the lifetime information and then it is not ensured that the temporary array [1.0] will live long enough, I believe. At least I faced this issue a few times already in prior rust/ash version. In release mode this tend to cause issues then as the memory gets reused.

Reply:
`build()` below is mandatory because we're populating an array of `DeviceQueueCreateInfo`s. The use of temporaries here and below is sound because temporaries [are guaranteed to outlive the statement they occur in](https://doc.rust-lang.org/reference/expressions.html#temporaries), which is the only context in which they are accessed.